conans/client/tools/win.py

```diff
@@ -14,10 +14,93 @@
 from conans.util.env_reader import get_env
 from conans.util.files import decode_text, save, mkdir_tmp
 from conans.unicode import get_cwd
+from conans.model.version import Version
 
 _global_output = None
 
 
+def _visual_compiler_cygwin(output, version):
+    if os.path.isfile("/proc/registry/HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/Windows/CurrentVersion/ProgramFilesDir (x86)"):
+        is_64bits = True
+    else:
+        is_64bits = False
+
+    if is_64bits:
+        key_name = r'HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\VisualStudio\SxS\VC7'
+    else:
+        key_name = r'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\VisualStudio\SxS\VC7'
+
+    if not os.path.isfile("/proc/registry/" + key_name.replace('\\', '/') + "/" + version):
+        return None
+
+    installed_version = Version(version).major(fill=False)
+    compiler = "Visual Studio"
+    output.success("CYGWIN: Found %s %s" % (compiler, installed_version))
+    return compiler, installed_version
+
+
+@contextmanager
+def _system_registry_key(key, subkey):
+    from six.moves import winreg  # @UnresolvedImport
+    hkey = winreg.OpenKey(key, subkey)
+    try:
+        yield hkey
+    finally:
+        winreg.CloseKey(hkey)
+
+
+def _visual_compiler(output, version):
+    """"version have to be 8.0, or 9.0 or... anything .0"""
+    if platform.system().startswith("CYGWIN"):
+        return _visual_compiler_cygwin(output, version)
+
+    if version == "15":
+        vs_path = os.getenv('vs150comntools')
+        path = vs_path or vs_installation_path("15")
+        if path:
+            compiler = "Visual Studio"
+            output.success("Found %s %s" % (compiler, "15"))
+            return compiler, "15"
+        return None
+
+    version = "%s.0" % version
+    from six.moves import winreg  # @UnresolvedImport
+    try:
+        try:
+            with _system_registry_key(winreg.HKEY_LOCAL_MACHINE,
+                                     r"SOFTWARE\Microsoft\Windows\CurrentVersion") as hKey:
+                winreg.QueryValueEx(hKey, "ProgramFilesDir (x86)")
+            is_64bits = True
+        except EnvironmentError:
+            is_64bits = False
+
+        if is_64bits:
+            key_name = r'SOFTWARE\Wow6432Node\Microsoft\VisualStudio\SxS\VC7'
+        else:
+            key_name = r'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\VisualStudio\SxS\VC7'
+
+        try:
+            with _system_registry_key(winreg.HKEY_LOCAL_MACHINE, key_name) as key:
+                winreg.QueryValueEx(key, version)
+
+            installed_version = Version(version).major(fill=False)
+            compiler = "Visual Studio"
+            output.success("Found %s %s" % (compiler, installed_version))
+            return compiler, installed_version
+        except EnvironmentError:
+            return None
+    except (WindowsError, EnvironmentError):
+        return None
+
+
+def latest_visual_studio_version_installed(output):
```

Comment:
This function should only return the version of Visual Studio as stated in its name

Reply:
I wouldn't oppose having the compiler as return if this was something internal, but as this is going to be a public function under the tools module I consider the compiler return is useless for the user

Reply:
changed tu return just version